rpc/src/rpc.rs

```diff
@@ -436,11 +472,18 @@ impl JsonRpcRequestProcessor {
         })?;
         let encoding = encoding.unwrap_or(UiAccountEncoding::Binary);
 
-        let response = get_encoded_account(&bank, pubkey, encoding, data_slice, None)?;
+        let response = self
+            .runtime
+            .spawn_blocking({
+                let bank = Arc::clone(&bank);
+                move || get_encoded_account(&bank, &pubkey, encoding, data_slice, None)
+            })
+            .await
+            .expect("rpc: get_encoded_account panicked")?;
```

Comment:
Let me say this back to you, just to make sure that I understand it correctly.

On first glance, creating a yield point here seems strange; there's a very little bit of work above and below, and this apparently synchronous call to `get_encoded_account()` in the middle. Why not just do all the work in one go?

However, if you hyperclick into `get_encoded_account()` long enough, you eventually bottom out at [this ‘read from disk’ call](https://github.com/anza-xyz/agave/blob/f3405d89eb0af318fd1a95ac24a459c5e5867d79/accounts-db/src/accounts_index/in_mem_accounts_index.rs#L374). This tells me that though this call appears sync, it can bottom out at an IOWait and cause the thread to go idle.

If `get_encoded_account()` and every function all the way down were `async` we would naturally get yieldyness through the use of `await`. Since it isn't, the change here introduces a yield-point in recognition that `get_encoded_account()` may wait on IO.

Reply:
All correct! I would add that there are other hidden costs in loading from accountsdb, especially if the account doesn't exist index lookup is very slow, and that _encoding_ is also very slow (can do compression etc) on large inputs and accounts can be up to 10MB.